### ISSUE TITLE
QuantityInfo can not be initialized for a custom quantity

### ISSUE THREAD

**Issue by rohahn** [angularsen/UnitsNet#895]
**Describe the bug**
A custom unit that is not located in the _UnitsNet_ assembly can not initialize a _QuantityInfo_ instance which is required to implement the _IQuantity_ interface.

The following code in _QuantityInfo_ assumes that all unit enumerations must be located in the same assembly as the _Length_ type (UnitsNet assembly), which is not valid for custom quantities.

```
        private static readonly Type[] UnitEnumTypes = typeof(Length)
            .Wrap()
            .Assembly
            .GetExportedTypes()
            .Where(t => t.Wrap().IsEnum && t.Namespace == UnitEnumNamespace && t.Name.EndsWith("Unit"))
            .ToArray(); 
```

Since _HowMuch.QuantityInfo_ is not called by any unit test, the problem is not exposed.

**To Reproduce**
New unit test for QuantityInfoTests:

        [Fact]
        public void Constructor_AssignsPropertiesForExternalQuantity()
        {
            var expectedZero = new HowMuch(10, HowMuchUnit.Some);
            var expectedUnitInfos = new UnitInfo[]
            {
                new UnitInfo(HowMuchUnit.Some, BaseUnits.Undefined),
                new UnitInfo(HowMuchUnit.ATon, BaseUnits.Undefined),
                new UnitInfo(HowMuchUnit.AShitTon, BaseUnits.Undefined)
            };
            var expectedBaseUnit = HowMuchUnit.Some;
            var expectedQuantityType = QuantityType.Undefined;
            var expectedBaseDimensions = BaseDimensions.Dimensionless;

            var info = new QuantityInfo(nameof(HowMuch), expectedUnitInfos,
                expectedBaseUnit, expectedZero, expectedBaseDimensions);

            Assert.Equal(expectedZero, info.Zero);
            Assert.Equal(nameof(HowMuch), info.Name);
            Assert.Equal(expectedUnitInfos, info.UnitInfos);
            Assert.Equal(expectedQuantityType, info.QuantityType);
            Assert.Equal(expectedBaseDimensions, info.BaseDimensions);

            // Obsolete members
            Assert.Equal( expectedBaseUnit, info.BaseUnit );
            Assert.Equal(new[] {nameof(HowMuchUnit.Some), nameof(HowMuchUnit.ATon), nameof(HowMuchUnit.AShitTon)}, info.UnitNames);
        }

**Expected behavior**
_QuantityInfo_ instances can be created for custom units.
_QuantityInfo_ should have a constructor with a Type parameter for the units enumeration, so the _UnitEnumTypes_ field and following line can be removed:
`UnitType = UnitEnumTypes.First(t => t.Name == $"{name}Unit");`

**Comment by tjrtfarnham** [comment]
@angularsen Is there any time frame when this might be fixed and available on nuget?  I am encountering the same problem.  I am trying to create a custom Quantity and add it to the Quantity.ByName dictionary so it can be used by UI using standard methods.  BTW - nice work with this library.

**Comment by angularsen** [comment]
@tjrtfarnham The PR was just merged and I have queued nuget 4.87.0. It should be out in not too long.

[Release UnitsNet/4.87.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/UnitsNet%2F4.87.0)
